src/injectHighlightWrappers.ts

```diff
@@ -80,9 +88,38 @@ function normalizeHighlights(highlights: Node[]) {
     }
   });
 
+  for (const [index, node] of (normalizedHighlights as HTMLElement[]).entries()) {
+    if (index === 0) {
+      node.classList.add('first')
+    }
+
+    if (hasBlockContent(node)) {
+      node.classList.add('block')
+    } else {
+      node.classList.add('text')
+    }
+
+    if (index === (normalizedHighlights.length - 1)) {
+      node.classList.add('last')
+    }
+  }
+
   return normalizedHighlights;
 }
 
+/**
+ * Check if there are block elements inside node.
+ * Block elements are: img, video and iframe.
+ * @param {HTMLElement} node
+ * @returns {boolean}
+ */
+function hasBlockContent(node: HTMLElement) {
+  for (const block of BLOCK_ELEMENTS) {
+    if (node.querySelector(block)) { return true; }
+  }
+  return false;
```

Comment:
this whole thing could be `return !!node.querySelector(BLOCK_ELEMENTS.join(','));`